packages/samples/headless-react/src/pages/SamlPage.tsx

```diff
@@ -1,41 +1,49 @@
-import {buildSamlClient, SamlClientOptions} from '@coveo/auth';
+import {buildSamlClient, SamlClient, SamlClientOptions} from '@coveo/auth';
 import {buildSearchEngine} from '@coveo/headless';
-import {useEffect, useState, PropsWithChildren, useMemo} from 'react';
+import {
+  useState,
+  useMemo,
+  useRef,
+  useEffect,
+  PropsWithChildren,
+  FunctionComponent,
+} from 'react';
 import {AppContext} from '../context/engine';
 
-export function SamlPage({
-  children,
-  ...samlClientOptions
-}: PropsWithChildren<SamlClientOptions>) {
-  const [initialAccessToken, setInitialAccessToken] = useState('');
-  const samlClient = useMemo(
-    () => buildSamlClient(samlClientOptions),
-    [samlClientOptions]
-  );
+const samlClientOptions: SamlClientOptions = {
+  organizationId: '',
+  provider: '',
+};
```

Comment:
Moved the options here, since the SAML client's `authenticate` method is deterministic and we don't want to suggest the possibility of changing the options of the SAML client.

Reply:
I could try to add a util for authentication that isn't deterministic, or I could update `@coveo/auth` (not sure what I'd do yet)